nni/compression/pytorch/compressor.py

```diff
@@ -411,17 +418,29 @@ def export_model(self, model_path, mask_path=None, onnx_path=None, input_shape=N
 
         torch.save(self.bound_model.state_dict(), model_path)
         _logger.info('Model state_dict saved to %s', model_path)
+
         if mask_path is not None:
             torch.save(mask_dict, mask_path)
             _logger.info('Mask dict saved to %s', mask_path)
+
         if onnx_path is not None:
-            assert input_shape is not None, 'input_shape must be specified to export onnx model'
-            # input info needed
-            if device is None:
-                device = torch.device('cpu')
-            input_data = torch.Tensor(*input_shape)
-            torch.onnx.export(self.bound_model, input_data.to(device), onnx_path)
-            _logger.info('Model in onnx with input shape %s saved to %s', input_data.shape, onnx_path)
+            assert input_shape is not None or dummy_input is not None,\
+                'input_shape or dummy_input must be specified to export onnx model'
+            # create dummy_input using input_shape if input_shape is not passed
+            if dummy_input is None:
+                if device is None:
+                    device = torch.device('cpu')
+                input_data = torch.Tensor(*input_shape).to(device)
+            else:
+                input_data = dummy_input
```

Comment:
what if user both set `dummy_input` and `device`

Reply:
I think device should be ignored in that case. Have updated the docstring.

Reply:
recommand we also `input_data = dummy_input.to(device)`, or this may confuse user, if user also set `device` but we ignore it.

Reply:
But I think this operation may fail when e.g., `dummy_input` is a tuple